plugin/pkg/admission/resourcequota/apis/resourcequota/types.go

```diff
@@ -54,4 +54,15 @@ type LimitedResource struct {
 	// with any storage class, the list would include
 	// ".storageclass.storage.k8s.io/requests.storage"
 	MatchContains []string
+
+	// For each intercepted request, the quota system will figure out if the input object
+	// satisfies a scope which is present in this listing, then
+	// quota system will ensure that there is a covering quota.  In the
+	// absence of a covering quota, the quota system will deny the request.
+	// For example, if an administrator wants to globally enforce that
+	// a quota must exist to create a pod with "cluster-services" priorityclass
+	// the list would include
+	// "PriorityClassNameIn=cluster-services"
+	// +optional
+	MatchScopes []string `json:"matchScopes,omitempty"`
```

Comment:
at this point, strongly coupling quota with priority is not the end of the world.

```
PrioritySelector struct {
  PriortySelectorTerms []PrioritySelectorTerm
}

ResourceQuotaSpec struct {
// scope quota to priority, resources must match pod names
Priority *PrioritySelector
... 
```

in the end, this may just be simpler for end users.